actuators/machine.go

```diff
@@ -132,18 +155,43 @@ func (m *Machine) Create(ctx context.Context, c *clusterv1.Cluster, machine *clu
 	}
 
 	fmt.Println("Creating a new worker node")
-	worker, err := actions.AddWorker(c.Name, machine.Spec.Versions.Kubelet)
+	worker, err := actions.AddWorker(c.Name, machine.GetName(), machine.Spec.Versions.Kubelet)
 	if err != nil {
 		fmt.Printf("%+v", err)
 		return err
 	}
-	name := providerID(worker.Name())
-	machine.Spec.ProviderID = &name
-	return m.save(old, machine)
+	providerID := providerID(worker.Name())
+	machine.Spec.ProviderID = &providerID
+	nodeUID, err := actions.GetNodeRefUID(c.GetName(), worker.Name())
+	if err != nil {
+		fmt.Printf("%+v", err)
+		return err
+	}
+	nodeRef := &apicorev1.ObjectReference{
+		Kind:       "Node",
+		APIVersion: apicorev1.SchemeGroupVersion.String(),
+		Name:       worker.Name(),
+		UID:        types.UID(nodeUID),
+	}
+	return m.save(old, machine, nodeRef)
 }
 
+// Delete returns nil when the machine no longer exists or when a successful delete has happened.
 func (m *Machine) Delete(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
-	return actions.DeleteNode(cluster.Name, providerNameToLookupID(*machine.Spec.ProviderID))
+	if machine.Status.NodeRef != nil {
```

Comment:
Maybe add the machine name to reflect with machine the messagesbelow refers to?

Reply:
I actually don't think this message is necessary, I'm going to nix it entirely